packages/app/src/sandbox/eval/transpilers/babel/worker/babel-worker.js

```diff
@@ -208,28 +215,33 @@ async function installPreset(Babel, BFSRequire, preset, currentPath, isV7) {
 
   let evaluatedPreset = null;
 
+  const presetName = await resolveDependencyName(preset, isV7, true);
+
   try {
+    await downloadPath(presetName);
     evaluatedPreset = evaluateFromPath(
       fs,
       BFSRequire,
-      preset,
+      presetName,
       currentPath,
       Babel.availablePlugins,
       Babel.availablePresets
     );
-  } catch (e) {
-    const prefixedName = getPrefixedPresetName(preset, isV7);
+  } catch (firstError) {
+    console.warn('First time compiling ' + preset + ' went wrong, got:');
+    console.warn(firstError);
 
-    evaluatedPreset = evaluateFromPath(
-      fs,
-      BFSRequire,
-      prefixedName,
-      currentPath,
-      Babel.availablePlugins,
-      Babel.availablePresets
-    );
+    /**
+     * We assume that a file is missing in the in-memory file system, and try to download it by
+     * parsing the error.
+     */
+    evaluatedPreset = await downloadFromError(firstError).then(() => {
+      resetCache();
+      return installPreset(Babel, BFSRequire, preset, currentPath, isV7);
+    });
   }
 
+  console.warn('Downloaded ' + preset);
```

Comment:
Should we leave this around or remove it?

Reply:
I'm thinking of deleting all of these anyway, or only show in dev. I'll make that change!

Reply:
Could be useful: [babel-plugin-transform-remove-console](https://www.npmjs.com/package/babel-plugin-transform-remove-console)